src/ploomber_core/config.py

```diff
@@ -112,19 +116,46 @@ def __setattr__(self, name, value):
             raise ValueError(f"{name} not a valid field")
         else:
             super().__setattr__(name, value)
-            self._write()
+
+            # Check if the filesystem is writable
+            if self.filesystem_writable() and name != "writable":
+                self._write()
 
     def load_config(self):
         config = None
-        path = self.path()
-        is_config_exist = Path.is_file(path)
-        if is_config_exist:
-            text = path.read_text()
-            if text:
-                config = yaml.safe_load(text)
+
+        if self.writable:
+            path = self.path()
+            is_config_exist = Path.is_file(path)
+            if is_config_exist:
+                text = path.read_text()
+                if text:
+                    config = yaml.safe_load(text)
 
         return config
 
+    def random_string(self, length=10):
```

Comment:
no need to create a separate method if we're only using this to test if the filesystem is writable, we can embed the logic in filesystem_writable